bin/check-uploaded-version.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+#
+# Usage: bin/check-uploaded-version.sh [PACKAGE...]
+#
+# Finds packages where the uploaded version of the package does not
+# match what is committed to the repository.
+#
+set -euo pipefail
+
+all_plans() {
+    for i in */plan.sh; do
+        dirname "$i"
+    done
+}
+
+clean_args() {
+    for arg in "$@"; do
+        arg=${arg%/}
+        if [[ "$arg" = */* ]]; then
+            dirname "$arg"
+        else
+            echo "$arg"
+        fi
+    done
+}
+
+get_plan_data() {
+    local plan_directory=$1
+    (
+        unset pkg_name pkg_version
+        set +u
+        set +e
+        export PLAN_CONTEXT="$plan_directory"
+        cd "$PLAN_CONTEXT"
+        # shellcheck disable=SC1091
+        . "plan.sh"
+        echo "$pkg_name ${pkg_version:-unknown}"
+    )
+}
+
+get_latest_stable() {
+    local pkg_name=$1
+    local url="https://bldr.habitat.sh/v1/depot/channels/core/stable/pkgs/$pkg_name/latest"
+    curl "$url" 2>/dev/null | jq -r .ident.version
+}
+
+for cmd in curl jq dirname; do
+    if ! command -v $cmd >/dev/null 2>&1; then
+        echo "error: $cmd required but not found"
+        exit 1
+    fi
+done
+
+declare -a plans_to_check
+if [[ "$#" -ne 0 ]]; then
+    mapfile -t plans_to_check < <(clean_args "$@")
```

Comment:
I tried running this inside a studio and on my local workstation (MacOS) and it failed to run `mapfile`:
```
➜  core-plans git:(ssd/check-uploaded-version) bin/check-uploaded-version.sh
bin/check-uploaded-version.sh: line 58: mapfile: command not found
```

I can install it but, should we check or install this command?

Reply:
It's not a command, it is a bash 4 feature.

Reply:
Im fine leaving it, as an idea, would it be nice to check if the binary exist and use it, and if it doesn't, use something else? 

If not, then just print out a requirements message? 🤔

Reply:
I've added a check for Bash 4 generally since most of the scripts we write in Habitat land assume various modern bash features.